mrp_unbuild_rebuild_variant/models/mrp_unbuild_rebuild_variant.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, exceptions, fields, models
+
+
+class MrpUnbuildRebuildVariant(models.Model):
+
+    _name = "mrp.unbuild.rebuild.variant"
+
+    name = fields.Char(
+        default=lambda self: self.env["ir.sequence"].next_by_code(self._name),
+        readonly=True,
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("done", "Done")],
+        required=True,
+        default="draft",
+        readonly=True,
+    )
+    # TODO: WHAT SHOULD BE COMPUTED HERE
+    unbuild_product_id = fields.Many2one("product.product")
+    unbuild_bom_id = fields.Many2one("mrp.bom", compute="_compute_unbuild_product_id")
+    unbuild_location_id = fields.Many2one("stock.location")
+    # if unbuild product is tracked:
+    unbuild_lot_id = fields.Many2one(
+        "stock.production.lot", domain="[('product_id', '=', unbuild_product_id)]"
+    )
+    # if unbuild product's bom includes tracked components:
+    unbuild_original_mo_id = fields.Many2one("mrp.production")
+    # if unbuild product's bom does not include tracked components:
+    quantity = fields.Float()
+
+    rebuild_product_template = fields.Many2one(
+        "product.template", compute="_compute_valid_template"
+    )
+
+    # TODO: Domain for same product.template
+    rebuild_product_id = fields.Many2one(
+        "product.product",
+        # FIXME: domain="[
+        # FIXME:    ('product_tmpl_id', '=', 'unbuild_product_id.product_tmpl_id.id')
+        # FIXME: ]"
+    )
+    # TODO: Check if this bom could be different than unbuild_bom_id
```

Comment:
For a first version, we could force to use the same mrp.bom for unbuild and rebuild because we don't have any handle to select potentially missing components from the stock.

Reply:
As I said above, I don't think we _need_ to handle that case.
We just need to force component lot for the those that both boms are sharing.
In such case, we force components lots. Otherwise, we let Odoo select components lots.
No?